src/AEOS/AEOSSyncTool/Adapters/SmartFaceDataAdapter.cs

```diff
@@ -0,0 +1,335 @@
+    {
+        private readonly ILogger logger;
+        private readonly IConfiguration configuration;
+        private readonly IHttpClientFactory httpClientFactory;
+        private readonly SmartFaceGraphQLClient graphQlClient;
+        private string SmartFaceURL;
+        private string SmartFaceGraphQL;
+        private int SmartFaceSetPageSize;
+        private int SmartFacePageSize;
+        private int SmartFaceDefaultThreshold;
+
+        private int MaxFaces;
+        private int MaxFaceSize;
+        private int MinFaceSize;
+        private int ConfidenceThreshold;
+        private bool KeepAutoLearnPhotos;
+        private Dictionary<string, bool> SmartFaceSyncedWatchlists = new();
+
+        private string AeosWatchlistName;
+
+        public SmartFaceDataAdapter(
+            ILogger logger,
+            IConfiguration configuration,
+            IHttpClientFactory httpClientFactory,
+            SmartFaceGraphQLClient graphQlClient
+
+        )
+        {
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this.configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+            this.httpClientFactory = httpClientFactory ?? throw new ArgumentNullException(nameof(httpClientFactory));
+            this.graphQlClient = graphQlClient ?? throw new ArgumentNullException(nameof(httpClientFactory));
+            this.logger.Debug("SmartFaceDataAdapter Initiated");
+
+            SmartFaceURL = configuration.GetValue<string>("aeossync:SmartFace:RestApi:ServerUrl");
+            SmartFaceGraphQL = configuration.GetValue<string>("aeossync:SmartFace:GraphQL:ServerUrl");
+            SmartFaceSetPageSize = configuration.GetValue<int>("aeossync:SmartFace:GraphQL:PageSize");
+            AeosWatchlistName = configuration.GetValue<string>("aeossync:SmartFace:Import:Watchlist");
+            SmartFacePageSize = configuration.GetValue<int>("aeossync:SmartFace:RestApi:PageSize");
+            SmartFaceDefaultThreshold = configuration.GetValue<int>("aeossync:SmartFace:Import:DefaultThreshold");
+            KeepAutoLearnPhotos = configuration.GetValue<bool>("AEOSSync:SmartFace:Import:KeepAutoLearnPhotos");
+            MaxFaces = configuration.GetValue<int>("AEOSSync:SmartFace:Import:FaceDetectorConfig:MaxFaces");
+            MaxFaceSize = configuration.GetValue<int>("AEOSSync:SmartFace:Import:FaceDetectorConfig:MaxFaceSize");
+            MinFaceSize = configuration.GetValue<int>("AEOSSync:SmartFace:Import:FaceDetectorConfig:MinFaceSize");
+            ConfidenceThreshold = configuration.GetValue<int>("AEOSSync:SmartFace:Import:FaceDetectorConfig:ConfidenceThreshold");
+            configuration.Bind("AEOSSync:SmartFace:Export:SyncedWatchlists", SmartFaceSyncedWatchlists);
+
+            if (SmartFaceURL == null)
+            {
+                throw new InvalidOperationException("The SmartFace URL is not read.");
+            }
```

Comment:
this can be easily onelined:

 SmartFaceURL = configuration.GetValue<string>("aeossync:SmartFace:RestApi:ServerUrl") ?? throw new InvalidOperationException("The SmartFace URL is not read.");

Reply:
And "...is not read" should rather be "...is not set/defined" IMHO

Reply:
Fixed across 2 projects.